---
title: "RNPC3 short-read minor class intron splicing analysis"
author: "Xueyi Dong"
date: '2022-07-07'
output:
  html_document:
    toc: true
    toc_depth: 3
    code_folding: show
    
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache = TRUE)
```

## Introduction

### Samples information

A549 cell line samples provided by Joan Heath.

The gene-level analysis for this dataset was done previously.

### Pre-processing

The transcriptome was detected by bambu on long-read data.

Reads were quantified against bambu transcripts using Salmon version 1.9.0 mapping-based mode.

### Aims and plan

We want to apply the same DTE and DTU pipeline that was used on long-read transcript-level count data.

## Prepare and load the data

```{r setup2}
library(edgeR)
library(limma)
library(Glimma)
library(ggplot2)
# library(superintronic)
# suppressPackageStartupMessages(library(plyranges))
```

```{r load_data}
tmap <- read.delim("../bambu/out/bambu_comp.extended_annotations.gtf.tmap")
mi_gene <- xlsx::read.xlsx("../HomoSapiens_IntronInfo.xlsx", sheetIndex = 1)
```

Counts were loaded using `catchSalmon` function from *edgeR*.

```{r catchSalmon}
counts <- catchSalmon(path = paste0("../salmon/short_bambu/", list.files("../salmon/short_bambu/")))
summary(counts$annotation$Overdispersion)
```

## Differential expression analysis

### Preparing data object

Raw counts were divided by over-dispersion coefficients.

```{r DGEList}
x <- DGEList(counts = counts$counts / counts$annotation$Overdispersion, genes = counts$annotation)
x$samples$group <- rep(c("control", "targeted"), c(4, 4))
colnames(x) <- paste(rep(c("NT", "RNPC3"), each = 4), rep(1:4, 2), sep = "_")
x$genes <- cbind(x$genes, tmap[match(rownames(x), tmap$qry_id), 1:6])
x$genes$isMinorIntron <- x$genes$ref_gene_id %in% mi_gene$Gene.Name
```
### Filtering, normalization and MDS plot

It may take a while to load the interactive MDS plot.
```{r preprocess}
filt <- filterByExpr(x)
table(filt)
x <- x[filt, ]
x <- calcNormFactors(x)
x$samples
glimmaMDS(x)
pdf("plots/shortDTE/MDS.pdf", height = 5, width = 8)
lcpm <- cpm(x, log=TRUE)
plotMDS(lcpm,
        col = rep(c("#c06636", "#646e3b"), each = 4),
        label = paste(rep(c("NT", "RNPC3"), each = 4), 1:4, sep = "-"),
        main = "Illumina short-read MDS plot",
        cex.main = 1.5, cex.lab = 1.5, cex.axis = 1.5, cex = 1.5)
dev.off()
pdf("plots/shortDTE/MDS_var.pdf", height = 5, width = 8)
# par(mfrow = c(1, 2))
layout(mat = matrix(1:2, ncol = 2), width = c(1.7, 1))
mds <- plotMDS(lcpm,
        col = rep(c("#c06636", "#646e3b"), each = 4),
        label = paste(rep(c("NT", "RNPC3"), each = 4), 1:4, sep = "-"),
        main = "Illumina short-read MDS plot",
        cex.main = 1.3, cex.lab = 1.3, cex.axis = 1.2,
        xlim = c(-1.9, 0.75))
barplot(mds$var.explained[1:7], names.arg = 1:7,
        xlab = "Dimension",
        ylab = "Proportion of variance explained",
        cex.main = 1.3, cex.lab = 1.3, cex.axis = 1.2,
        ylim = c(0, 0.25))
dev.off()
```

On the MDS plot, sample NT_4 and RNPC3_4 performed slightly different from other samples. Similar situation were also observed in our long-read data.

### DE analysis using *edgeR* quasi-likelihood method

```{r DE}
design <- model.matrix(~x$samples$group)
x <- estimateDisp(x, design)
plotBCV(x)
pdf("plots/shortDTE/BCV.pdf", height = 5, width = 8)
plotBCV(x,
        cex.main = 1.5, cex.lab = 1.5, cex.axis = 1.5,
        main = "Illumina short-read BCV")
dev.off()
# ggplot version of BCV plot
dispersion <- data.frame(
  AveLogCPM = x$AveLogCPM,
  common.dispersion = x$common.dispersion,
  trended.dispersion = x$trended.dispersion,
  tagwise.dispersion = x$tagwise.dispersion
)
dispersion <- dispersion[order(dispersion$AveLogCPM),]
pdf("plots/shortDTE/BCV_gg.pdf", height = 5, width = 8)
bcv.plot <- ggplot(dispersion)+
  stat_binhex(aes(x = AveLogCPM, y = sqrt(tagwise.dispersion)), bins = 100) +
  scale_fill_viridis(option = "magma", limits=c(0, 450)) +
  # geom_hline(yintercept = sqrt(x$common.dispersion), aes(colour = "Common")) +
  geom_line(aes(x = AveLogCPM, y = sqrt(common.dispersion), colour = "Common")) +
  geom_line(aes(x = AveLogCPM, y = sqrt(trended.dispersion), colour = "Trend")) +
  scale_colour_manual(name = "BCV",
                      breaks = c("Common", "Trend"),
                      values = c("Common" = "red", "Trend" = "blue")) +
  theme_bw() +
  labs(x = "Average log CPM", y = "Bioligical coefficient of variation",
       fill = "Tagwise BCV\ncount",
       title = "Illumina short-read BCV") +
  theme(text = element_text(size = 16))
print(bcv.plot)
dev.off()
saveRDS(bcv.plot, "shortBCV.RDS")
qlfit <- glmQLFit(x, design)
plotQLDisp(qlfit)
res <- glmQLFTest(qlfit)
plotMD(res, main = "RNPC3-targeted vs NT")
pdf("plots/shortDTE/MD.pdf", height = 5, width = 8)
plotMD(res, 
       cex.main = 1.5, cex.lab = 1.5, cex.axis = 1.5,
       main = "Illumina RNPC3-targeted vs NT")
dev.off()
# number of DE transcripts
summary(decideTestsDGE(res))
# top DE transcripts
topTags(res)
tt <- topTags(res, n=Inf)
write.csv(tt, "results_short/DE_topTags.csv")

# check number of MI transcripts in top n DTEs
top_n_mi <- function(ts){
  res = data.frame(
    n = 1:min(sum(ts$FDR < 0.05), sum(ts$isMinorIntron))
  )
  res$n_mi = sapply(res$n, function(x){
    sum(ts[1:x, "isMinorIntron"])
  }, simplify=TRUE)
  return(res)
}
topN.tt <- top_n_mi(as.data.frame(tt))
pdf("plots/shortDTE/topN_MI_DTE.pdf", height = 5, width = 6)
plot.topN.DTE <- ggplot(topN.tt, aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DE transcripts", y = "number of minor intron transcripts",
       title = "Illumina DTEs") +
  theme_bw() +
  theme(text = element_text(size = 18))
print(plot.topN.DTE)
dev.off()

pdf("plots/shortDTE/topN_MI_DTE_top50.pdf", height = 5, width = 4)
plot.topN.DTE.top50 <- ggplot(topN.tt[1:50, ], aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DE transcripts", y = "number of minor intron transcripts") +
  theme_bw() +
  theme(text = element_text(size = 18))
print(plot.topN.DTE.top50)
dev.off()

pdf("plots/shortDTE/topN_MI_DTE_top100.pdf", height = 5, width = 4)
plot.topN.DTE.top100 <- ggplot(topN.tt[1:100, ], aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DE transcripts", y = "number of minor intron transcripts") +
  theme_bw() +
  theme(text = element_text(size = 18))
print(plot.topN.DTE.top100)
dev.off()

pdf("plots/shortDTE/topN_MI_DTE_combined.pdf", height = 5, width = 10)
ranges = c(layer_scales(plot.topN.DTE.top100)$x$range$range,
           layer_scales(plot.topN.DTE.top100)$y$range$range)
plot_grid(plot.topN.DTE +
            annotate("rect", 
                     xmin = ranges[1] - (ranges[2] - ranges[1]) * 0.05,
                     xmax = ranges[2] + (ranges[2] - ranges[1]) * 0.05,
                     ymin = ranges[3] - (ranges[4] - ranges[3]) * 0.05,
                     ymax = ranges[4] + (ranges[4] - ranges[3]) * 0.05,
                     alpha = .3,
                     fill = "blue"), 
          plot.topN.DTE.top100 +
            theme(axis.title.y = element_blank()),
          rel_widths = c(6, 4))
dev.off()

pdf("plots/shortDTE/topN_MI_DTE_new.pdf", height = 5, width = 8)
print(plot.topN.DTE +
            annotate("rect", 
                     xmin = ranges[1] - (ranges[2] - ranges[1]) * 0.05,
                     xmax = ranges[2] + (ranges[2] - ranges[1]) * 0.05,
                     ymin = ranges[3] - (ranges[4] - ranges[3]) * 0.05,
                     ymax = ranges[4] + (ranges[4] - ranges[3]) * 0.05,
                     alpha = .2,
                     fill = "blue"))
dev.off()

pdf("plots/shortDTE/topN_MI_DTE_top100_small.pdf", height = 2, width = 3)
print(plot.topN.DTE.top100 +
            theme(axis.title.y = element_blank(),
                  axis.title.x = element_blank(),
                  panel.background = element_rect(fill = "#CCCCFF"),
                  panel.grid.major = element_blank(),
                  panel.grid.minor = element_blank()))
dev.off()
```

[DE results table](results_short/DE_topTags.csv)

[Number of minor intron genes in top N DE transcripts](plots_short/topN_MI_DTE.pdf)

### Gene set testing

We used the function `fry` from *limma* to test whether specific sets of transcripts are differentially expressed, and barcode plots were made to visualize the results.

The `PValue` column gives the two-sided p-value for testing whether the set is DE as a whole, either up or down. The `PValue.Mixed` column gives a p-value for testing whether genes in the set tend to be DE, without regard to direction. 

#### Minor intron genes from the list provided by Joan.

```{r geneSetMI}
lcpm <- cpm(x, log = TRUE)
res.fry <- fry(lcpm, x$genes$isMinorIntron,design)
res.fry
# barcodeplot(res$table$logFC, index = x$genes$isMinorIntron,
#             xlab = "log fold change",
#             main = paste0("Illumina minor intron genes, n=",
#                           res.fry$NGenes,
#                           "\nFry mixed p-value=",
#                           signif(res.fry$PValue.Mixed, 3)))
pdf("plots/shortDTE/barcode_minorIntron.pdf", height = 5, width = 20)
barcodeplot(res$table$logFC, index = x$genes$isMinorIntron,
            xlab = "log fold change",
            main = paste0("Illumina minor intron genes, n=",
                          res.fry$NGenes,
                          "\nFry mixed p-value=",
                          signif(res.fry$PValue.Mixed, 3)),
            cex.lab = 1.5,
            cex.main = 1.5,
            cex = 2)
dev.off()
pdf("plots/shortDTE/barcode_minorIntron_small.pdf", height = 5, width = 8)
barcodeplot(res$table$logFC, index = x$genes$isMinorIntron,
            xlab = "log fold change",
            main = paste0("Illumina minor intron genes, n=",
                          res.fry$NGenes,
                          "\nFry mixed p-value=",
                          signif(res.fry$PValue.Mixed, 3)),
            cex.lab = 1.5,
            cex.main = 1.5,
            cex = 2)
dev.off()
```

#### Other lists provided by Joan

Extract table from pdf and add information to`x$genes`
```{r geneLists}
library(tabulizer)
minor_ir <- as.data.frame(extract_tables("../metadata/Short-read splicing analysis.pdf", pages = 1)[[1]])
colnames(minor_ir) <- minor_ir[1 ,]
minor_ir <- minor_ir[-1, ]
colnames(minor_ir)[6:10] <- colnames(minor_ir)[5:9]
colnames(minor_ir)[4] <- "Coord start"
colnames(minor_ir)[5] <- "Coord end"
x$genes$isMinorIR <- x$genes$ref_gene_id %in% minor_ir$`Gene Name`

minor_as <- as.data.frame(extract_tables("../metadata/Short-read splicing analysis.pdf", pages = 2)[[1]])
colnames(minor_as) <- minor_as[1, ]
minor_as <- minor_as[-1, ]
minor_as <- minor_as[, -9]
colnames(minor_as)[5:9] <- colnames(minor_as)[4:8]
colnames(minor_as)[3] <- "Coord start" 
colnames(minor_as)[4] <- "Coord end"
minor_as$gene <- strsplit2(minor_as$`AS event – Gene ID – Gene Name`, "-")[,3]
x$genes$isMinorAS <- x$genes$ref_gene_id %in% minor_as$gene

ir_finder <- as.data.frame(extract_tables("../metadata/Short-read splicing analysis.pdf", pages = 3)[[1]])
colnames(ir_finder) <- ir_finder[1, ]
ir_finder <- ir_finder[-1, ]
x$genes$isIrFinder <- x$genes$ref_gene_id %in% ir_finder$Gene
```

Gene set testing

```{r geneSetLists}
# minor IR
res.fry <- fry(lcpm, x$genes$isMinorIR,design)
res.fry
# barcodeplot(res$table$logFC, index = x$genes$isMinorIR,
#             xlab = "log fold change",
#             main = paste0("Illumina Minor IR top 40 genes, n=",
#                           res.fry$NGenes,
#                           "\nFry mixed p-value=",
#                           signif(res.fry$PValue.Mixed, 3)))
pdf("plots/shortDTE/barcode_MinorIR.pdf", height = 5, width = 8)
barcodeplot(res$table$logFC, index = x$genes$isMinorIR,
            xlab = "log fold change",
            main = paste0("Illumina Minor IR top 40 genes, n=",
                          res.fry$NGenes,
                          "\nFry mixed p-value=",
                          signif(res.fry$PValue.Mixed, 3)),
            cex.lab = 1.5,
            cex.main = 1.5,
            cex = 2)
dev.off()
# minor AS
res.fry <- fry(lcpm, x$genes$isMinorAS,design)
res.fry
# barcodeplot(res$table$logFC, index = x$genes$isMinorAS,
#             xlab = "log fold change",
#             main = paste0("Illumina Minor AS top 40 gene, n=",
#                           res.fry$NGenes,
#                           "\nFry mixed p-value=",
#                           signif(res.fry$PValue.Mixed, 3)))
pdf("plots/shortDTE/barcode_minorAS.pdf", height = 5, width = 8)
barcodeplot(res$table$logFC, index = x$genes$isMinorAS,
            xlab = "log fold change",
            main = paste0("Illumina Minor AS significant genes, n=",
                          res.fry$NGenes,
                          "\nFry mixed p-value=",
                          signif(res.fry$PValue.Mixed, 3)),
            cex.lab = 1.5,
            cex.main = 1.5,
            cex = 2)
dev.off()

# IR finder
res.fry <- fry(lcpm, x$genes$isIrFinder,design)
res.fry
# barcodeplot(res$table$logFC, index = x$genes$isIrFinder,
#             xlab = "log fold change",
#             main = paste0("Illumina IR finder unbiased top 40 genes, n=",
#                           res.fry$NGenes,
#                           "\nFry mixed p-value=",
#                           signif(res.fry$PValue.Mixed, 3)))
pdf("plots/shortDTE/barcode_IRFinder.pdf", height = 5, width = 8)
barcodeplot(res$table$logFC, index = x$genes$isIrFinder,
            xlab = "log fold change",
            main = paste0("Illumina IR finder unbiased top 40 genes, n=",
                          res.fry$NGenes,
                          "\nFry mixed p-value=",
                          signif(res.fry$PValue.Mixed, 3)),
            cex.lab = 1.5,
            cex.main = 1.5,
            cex = 2)
dev.off()
```

#### Transcript classifications

Class "m" transripts
```{r geneSetM}
fry(lcpm, x$genes$class_code=="m", design)
barcodeplot(res$table$logFC, index = x$genes$class_code=="m",
            main = "Transcript class m, targeted vs control")
```


Class "n" transcipts
```{r geneSetN}
fry(lcpm, x$genes$class_code=="n", design)
barcodeplot(res$table$logFC, index = x$genes$class_code=="n",
            main = "Transcript class n, targeted vs control")
```

Class "j" transcripts
```{r geneSetJ}
fry(lcpm, x$genes$class_code=="j", design)
barcodeplot(res$table$logFC, index = x$genes$class_code=="j",
            main = "Transcript class j, targeted vs control")
```

## Differential transcript usage analysis

### DTU analysis using *limma-diffSplice*

```{r DTU}
v <- voom(x, design, plot = TRUE)
fit <- lmFit(v)
ds <- diffSplice(fit, geneid = x$genes$qry_gene_id, exonid = x$genes$qry_id)
# top DTU genes
topSplice(ds)
# top DTU transcripts
topSplice(ds, test = "t")
ts <- topSplice(ds, test = "simes", number = Inf)
ts.tx <- topSplice(ds, test = "t", number = Inf)
# number of DTU genes
table(ts$FDR < 0.05)
# number of DTU transcripts
table(ts.tx$FDR < 0.05)

write.csv(ts, file = "results_short/DTU_gene_topSplice.csv")
write.csv(ts.tx, file = "results_short/DTU_transcript_topSplice.csv")

# check number of top DTU MI genes

topN.ts <- top_n_mi(ts)
pdf("plots/shortDTU/topN_MI_DTU.pdf", height = 5, width = 4)
ggplot(topN.ts, aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DTU genes", y = "number of minor intron genes",
       title = "Illumina DTU genes") +
  theme_bw() +
  theme(text = element_text(size = 18))
dev.off()

topN.ts.tx <- top_n_mi(ts.tx)
pdf("plots/shortDTU/topN_MI_DTU_tx.pdf", height = 5, width = 4)
ggplot(topN.ts.tx, aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DTU transcripts", y = "number of minor intron transcripts",
       title = "Illumina DTU transcripts") +
  theme_bw() +
  theme(text = element_text(size = 18))
dev.off()
```

[DTU gene-level results table](results_short/DTU_gene_topSplice.csv)

[DTU transcript-level results table](results_short/DTU_transcript_topSplice.csv)

[Number of minor intron genes in top N DTU genes](plots_short/topN_MI_DTU.pdf)

[Number of minor intron transcripts in top N DTU transcripts](plots_short/topN_MI_DTU_tx.pdf)

### Gene set testing based on transcript-level DTU results

#### Minor intron genes

```{r geneSetDTUMI}
barcodeplot(ds$t[, 2], index = ds$genes$isMinorIntron,
            main = "Minor intron genes, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$isMinorIntron, ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$isMinorIntron, ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$isMinorIntron, ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value

pdf("plots/shortDTU/barcode_minorIntron.pdf", height = 5, width = 16)
barcodeplot(ds$t[, 2], index = ds$genes$isMinorIntron,
            main = paste0("Illumina transcripts of minor intron genes, n=",
                          sum(ds$genes$isMinorIntron),
                          "\nFisher's exact test p-value=",
                          signif(test.mixed$p.value, 3)),
            xlab = "t-statistics",
            cex.lab = 1.5,
            cex.main = 1.5)
dev.off()
pdf("plots/shortDTU/barcode_minorIntron_small.pdf", height = 5, width = 8)
barcodeplot(ds$t[, 2], index = ds$genes$isMinorIntron,
            main = paste0("Illumina transcripts of minor intron genes, n=",
                          sum(ds$genes$isMinorIntron),
                          "\nFisher's exact test p-value=",
                          signif(test.mixed$p.value, 3)),
            xlab = "t-statistics",
            cex.lab = 1.5,
            cex.main = 1.5,
            cex = 2)
dev.off()
```

#### Other lists provided by Joan

```{r geneSetDTUminorIR}
test.mixed <- fisher.test(table(ts.tx$isMinorIR, ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$isMinorIR, ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$isMinorIR, ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value

pdf("plots/shortDTU/barcode_minorIR.pdf", height = 5, width = 8)
barcodeplot(ds$t[, 2], index = ds$genes$isMinorIR,
            main = paste0("Illumina transcripts of minor IR top 40 genes, n=",
                          sum(ds$genes$isMinorIR),
                          "\nFisher's exact test p-value=",
                          signif(test.mixed$p.value, 3)),
            xlab = "t-statistics",
            cex.lab = 1.5,
            cex.main = 1.5)
dev.off()
```

```{r geneSetDTUminorAS}
test.mixed <- fisher.test(table(ts.tx$isMinorAS, ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$isMinorAS, ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$isMinorAS, ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
pdf("plots/shortDTU/barcode_minorAS.pdf", height = 5, width = 8)
barcodeplot(ds$t[, 2], index = ds$genes$isMinorAS,
             main = paste0("Illumina transcripts of minor AS significant genes, n=",
                          sum(ds$genes$isMinorAS),
                          "\nFisher's exact test p-value=",
                          signif(test.mixed$p.value, 3)),
            xlab = "t-statistics",
            cex.lab = 1.5,
            cex.main = 1.5)
dev.off()
```

```{r geneSetDTUirFinder}
test.mixed <- fisher.test(table(ts.tx$isIrFinder, ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$isIrFinder, ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$isIrFinder, ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
pdf("plots/shortDTU/barcode_IRFinder.pdf", height = 5, width = 8)
barcodeplot(ds$t[, 2], index = ds$genes$isIrFinder,
           main = paste0("Illumina transcripts of IR Finder top 40 genes, n=",
                          sum(ds$genes$isIrFinder),
                          "\nFisher's exact test p-value=",
                          signif(test.mixed$p.value, 3)),
            xlab = "t-statistics",
            cex.lab = 1.5,
            cex.main = 1.5)
dev.off()
```

#### Transcript classifications

Class "m" transripts
```{r geneSetDTUM}
barcodeplot(ds$t[, 2], index = ds$genes$class_code=="m",
            main = "Transcript class m, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$class_code=="m", ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$class_code=="m", ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$class_code=="m", ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```


Class "n" transcipts
```{r geneSetDTUN}
barcodeplot(ds$t[, 2], index = ds$genes$class_code=="n",
            main = "Transcript class n, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$class_code=="n", ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$class_code=="n", ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$class_code=="n", ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```

Class "j" transcripts
```{r geneSetDTUJ}
barcodeplot(ds$t[, 2], index = ds$genes$class_code=="j",
            main = "Transcript class j, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$class_code=="j", ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$class_code=="j", ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$class_code=="j", ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```


<!-- ## Visualization of transcripts coverage -->

<!-- Due to high computation load, visualization was done in a separate script. notice that genes without known strand information are not plotted. -->

```{r saveGeneName}
# save DTE gene names
tt <- as.data.frame(tt)
saveRDS(unique(tt$qry_gene_id[tt$FDR < 0.05 & tt$isMinorIntron]), "results_short/DTEgenesMI.RDS")
saveRDS(unique(tt$qry_gene_id[tt$FDR < 0.05 & !tt$isMinorIntron]), "results_short/DTEgenesNotMI.RDS")
saveRDS(unique(ts$qry_gene_id[ts$FDR < 0.05 & ts$isMinorIntron]), "results_short/DTUgenesMI.RDS")
saveRDS(unique(ts$qry_gene_id[ts$FDR < 0.05 & !ts$isMinorIntron]), "results_short/DTUgenesNotMI.RDS")
```

```{r txViz, eval=FALSE}
source("plot_cov.R")
```

<!-- [DTE minor intron genes](plots/cov_DTEgenesMI2.pdf) -->

<!-- [DTE not minor intron genes](plots/cov_DTEgenesNotMI2.pdf) -->

<!-- [DTU minor intron genes](plots/cov_DTUgenesMI2.pdf) -->

<!-- [DTU not minor intron genes](plots/cov_DTUgenesNotM2I.pdf) -->

<!-- [Minor intron genes that are not DTE or DTU but containing class m, n or j transcripts](plots/cov_notSigMI_mnj.pdf) -->

## To do list

1. Quality control plots

2. Compare results from long vs short read (roast gene set test, overlap etc.)

3. Try to see what genes made replicate 4 outliers. Can try robust eBayes and check prior df.

## Session information
```{r sessionInfo}
sessionInfo()
```
