---
title: "RNPC3 short-read minor class intron splicing analysis"
author: "Xueyi Dong"
date: '2022-07-07'
output:
  html_document:
    toc: true
    toc_depth: 3
    code_folding: show
    
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache = TRUE)
```

## Introduction

### Samples information

A549 cell line samples provided by Joan Heath.

The gene-level analysis for this dataset was done previously.

### Pre-processing

The transcriptome was detected by bambu on long-read data.

Reads were quantified against bambu transcripts using Salmon version 1.9.0 mapping-based mode.

### Aims and plan

We want to apply the same DTE and DTU pipeline that was used on long-read transcript-level count data.

## Prepare and load the data

```{r setup2}
library(edgeR)
library(limma)
library(Glimma)
library(ggplot2)
# library(superintronic)
# suppressPackageStartupMessages(library(plyranges))
```

```{r load_data}
tmap <- read.delim("../bambu/out/bambu_comp.extended_annotations.gtf.tmap")
mi_gene <- xlsx::read.xlsx("../HomoSapiens_IntronInfo.xlsx", sheetIndex = 1)
```

Counts were loaded using `catchSalmon` function from *edgeR*.

```{r catchSalmon}
counts <- catchSalmon(path = paste0("../salmon/short_bambu/", list.files("../salmon/short_bambu/")))
```

## Differential expression analysis

### Preparing data object

Raw counts were divided by over-dispersion coefficients.

```{r DGEList}
x <- DGEList(counts = counts$counts / counts$annotation$Overdispersion, genes = counts$annotation)
x$samples$group <- rep(c("control", "targeted"), c(4, 4))
colnames(x) <- paste(rep(c("NT", "RNPC3"), each = 4), rep(1:4, 2), sep = "_")
x$genes <- cbind(x$genes, tmap[match(rownames(x), tmap$qry_id), 1:6])
x$genes$isMinorIntron <- x$genes$ref_gene_id %in% mi_gene$Gene.Name
```
### Filtering, normalization and MDS plot

It may take a while to load the interactive MDS plot.
```{r preprocess}
filt <- filterByExpr(x)
table(filt)
x <- x[filt, ]
x <- calcNormFactors(x)
x$samples
glimmaMDS(x)
```

On the MDS plot, sample NT_4 and RNPC3_4 performed slightly different from other samples. Similar situation were also observed in our long-read data.

### DE analysis using *edgeR* quasi-likelihood method

```{r DE}
design <- model.matrix(~x$samples$group)
x <- estimateDisp(x, design)
plotBCV(x)
qlfit <- glmQLFit(x, design)
plotQLDisp(qlfit)
res <- glmQLFTest(qlfit)
plotMD(res, main = "RNPC3-targeted vs NT")
# number of DE transcripts
summary(decideTestsDGE(res))
# top DE transcripts
topTags(res)
tt <- topTags(res, n=Inf)
write.csv(tt, "results_short/DE_topTags.csv")

# check number of MI transcripts in top n DTEs
top_n_mi <- function(ts){
  res = data.frame(
    n = 1:min(sum(ts$FDR < 0.05), sum(ts$isMinorIntron))
  )
  res$n_mi = sapply(res$n, function(x){
    sum(ts[1:x, "isMinorIntron"])
  }, simplify=TRUE)
  return(res)
}
topN.tt <- top_n_mi(as.data.frame(tt))
pdf("plots_short/topN_MI_DTE.pdf", height = 4)
ggplot(topN.tt, aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DE transcripts", y = "number of minor intron transcripts")
dev.off()
```

[DE results table](results_short/DE_topTags.csv)

[Number of minor intron genes in top N DE transcripts](plots_short/topN_MI_DTE.pdf)

### Gene set testing

We used the function `fry` from *limma* to test whether specific sets of transcripts are differentially expressed, and barcode plots were made to visualize the results.

The `PValue` column gives the two-sided p-value for testing whether the set is DE as a whole, either up or down. The `PValue.Mixed` column gives a p-value for testing whether genes in the set tend to be DE, without regard to direction. 

#### Minor intron genes from the list provided by Joan.

```{r geneSetMI}
lcpm <- cpm(x, log = TRUE)
fry(lcpm, x$genes$isMinorIntron,design)
barcodeplot(res$table$logFC, index = x$genes$isMinorIntron,
            main = "Minor intron genes, targeted vs control")
```

#### Transcript classifications

Class "m" transripts
```{r geneSetM}
fry(lcpm, x$genes$class_code=="m", design)
barcodeplot(res$table$logFC, index = x$genes$class_code=="m",
            main = "Transcript class m, targeted vs control")
```


Class "n" transcipts
```{r geneSetN}
fry(lcpm, x$genes$class_code=="n", design)
barcodeplot(res$table$logFC, index = x$genes$class_code=="n",
            main = "Transcript class n, targeted vs control")
```

Class "j" transcripts
```{r geneSetJ}
fry(lcpm, x$genes$class_code=="j", design)
barcodeplot(res$table$logFC, index = x$genes$class_code=="j",
            main = "Transcript class n, targeted vs control")
```

## Differential transcript usage analysis

### DTU analysis using *limma-diffSplice*

```{r DTU}
v <- voom(x, design, plot = TRUE)
fit <- lmFit(v)
ds <- diffSplice(fit, geneid = x$genes$qry_gene_id, exonid = x$genes$qry_id)
# top DTU genes
topSplice(ds)
# top DTU transcripts
topSplice(ds, test = "t")
ts <- topSplice(ds, test = "simes", number = Inf)
ts.tx <- topSplice(ds, test = "t", number = Inf)
# number of DTU genes
table(ts$FDR < 0.05)
# number of DTU transcripts
table(ts.tx$FDR < 0.05)

write.csv(ts, file = "results_short/DTU_gene_topSplice.csv")
write.csv(ts.tx, file = "results_short/DTU_transcript_topSplice.csv")

# check number of top DTU MI genes

topN.ts <- top_n_mi(ts)
pdf("plots_short/topN_MI_DTU.pdf", height = 4)
ggplot(topN.ts, aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DTU genes", y = "number of minor intron genes")
dev.off()

topN.ts.tx <- top_n_mi(ts.tx)
pdf("plots_short/topN_MI_DTU_tx.pdf", height = 4)
ggplot(topN.ts.tx, aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DTU transcripts", y = "number of minor intron transcripts")
dev.off()
```

[DTU gene-level results table](results_short/DTU_gene_topSplice.csv)

[DTU transcript-level results table](results_short/DTU_transcript_topSplice.csv)

[Number of minor intron genes in top N DTU genes](plots_short/topN_MI_DTU.pdf)

[Number of minor intron transcripts in top N DTU transcripts](plots_short/topN_MI_DTU_tx.pdf)

### Gene set testing based on transcript-level DTU results

#### Minor intron genes

```{r geneSetDTUMI}
barcodeplot(ds$t[, 2], index = ds$genes$isMinorIntron,
            main = "Minor intron genes, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$isMinorIntron, ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$isMinorIntron, ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$isMinorIntron, ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```
#### Transcript classifications

Class "m" transripts
```{r geneSetDTUM}
barcodeplot(ds$t[, 2], index = ds$genes$class_code=="m",
            main = "Transcript class m, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$class_code=="m", ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$class_code=="m", ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$class_code=="m", ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```


Class "n" transcipts
```{r geneSetDTUN}
barcodeplot(ds$t[, 2], index = ds$genes$class_code=="n",
            main = "Transcript class n, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$class_code=="n", ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$class_code=="n", ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$class_code=="n", ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```

Class "j" transcripts
```{r geneSetDTUJ}
barcodeplot(ds$t[, 2], index = ds$genes$class_code=="j",
            main = "Transcript class j, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$class_code=="j", ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$class_code=="j", ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$class_code=="j", ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```


<!-- ## Visualization of transcripts coverage -->

<!-- Due to high computation load, visualization was done in a separate script. notice that genes without known strand information are not plotted. -->

```{r saveGeneName}
# save DTE gene names
tt <- as.data.frame(tt)
saveRDS(unique(tt$qry_gene_id[tt$FDR < 0.05 & tt$isMinorIntron]), "results_short/DTEgenesMI.RDS")
saveRDS(unique(tt$qry_gene_id[tt$FDR < 0.05 & !tt$isMinorIntron]), "results_short/DTEgenesNotMI.RDS")
saveRDS(unique(ts$qry_gene_id[ts$FDR < 0.05 & ts$isMinorIntron]), "results_short/DTUgenesMI.RDS")
saveRDS(unique(ts$qry_gene_id[ts$FDR < 0.05 & !ts$isMinorIntron]), "results_short/DTUgenesNotMI.RDS")
```

```{r txViz, eval=FALSE}
source("plot_cov.R")
```

<!-- [DTE minor intron genes](plots/cov_DTEgenesMI2.pdf) -->

<!-- [DTE not minor intron genes](plots/cov_DTEgenesNotMI2.pdf) -->

<!-- [DTU minor intron genes](plots/cov_DTUgenesMI2.pdf) -->

<!-- [DTU not minor intron genes](plots/cov_DTUgenesNotM2I.pdf) -->

<!-- [Minor intron genes that are not DTE or DTU but containing class m, n or j transcripts](plots/cov_notSigMI_mnj.pdf) -->


## To do list

1. Quality control plots

2. Compare results from long vs short read (roast gene set test, overlap etc.)

3. GO/KEGG analysis (the `goana` and `kegga` functions are only available for gene level)

## Session information
```{r sessionInfo}
sessionInfo()
```
