---
title: "RNPC3 minor intron splicing analysis"
author: "Xueyi Dong"
date: '2022-06-06'
output:
  html_document:
    toc: true
    toc_depth: 3
    code_folding: show
    
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache = TRUE)
```

## Introduction

### Samples information

A549 cell line samples provided by Joan Heath, sequenced using nanopore PCR cDNA protocol on 3 flow cells.

barcode01-04: controls (non-targeted), barcode05-08: targeted with rnpc3 siRNA 18.

### Pre-processing

Basecalling and demultiplexing were done using Guppy version 6.0.6.

Reads were mapped to human CHM13v2 reference genome using minimap2 version 2.17.

Novel isoforms identified and transcript-level counts generated using bambu version 1.0.3.

The gtf of identified isoforms was then compared to CHM13v2 annotation using gffcompare.

(Not finished) Quality control report of identified isoforms was generated using SQANTI3 version 5.0, with short read data to calculate splicing junction coverage.

### Aims and plan

We want to explore human minor introns and splicing events using long read sequencing.

First, we will check the novel isoforms detected. We will especially focus on whether there is "intron retention" class novel isoforms from the gene list provided by Joan and Stephen. 

Next, we will perform differential transcript expression (DTE) analysis and differential transcript usage (DTU) analysis to see what is changed between targeted and untargetet samples. Then, we will perform gene set testing to further explore the DTE and DTU results.

Finally, we will visualize the isoform structure and genomic coverage of interesting genes.

## Prepare and load the data

```{r setup2}
library(edgeR)
library(limma)
library(Glimma)
library(ggplot2)
library(superintronic)
suppressPackageStartupMessages(library(plyranges))
library(dplyr)
library(cowplot)
```

```{r load_data}
counts <- read.delim("../bambu/out/counts_transcript.txt")
isoClass <- read.delim("../bambu/sqanti_out/extended_annotations.genes.edited_classification.txt")
rownames(counts) <- counts$TXNAME
tmap <- read.delim("../bambu/out/bambu_comp.extended_annotations.gtf.tmap")
mi_gene <- xlsx::read.xlsx("../HomoSapiens_IntronInfo.xlsx", sheetIndex = 1)
```

## Isoform classification

```{r isoClass}
isoClass$count <- rowSums(counts[,c(-1, -2)])[match(isoClass$isoform, counts$TXNAME)]
# check number of zero count isoforms
table(isoClass$count == 0)
# check number of low count isoforms
table(isoClass$count < 10)
isoClass$structural_category[isoClass$structural_category %in% c("genic", "genic_intron", "intergenic")] <- "genic/intergenic"
isoClass$structural_category <- factor(isoClass$structural_category, levels =c("full-splice_match", "incomplete-splice_match", "novel_in_catalog", "novel_not_in_catalog", "genic/intergenic", "fusion", "antisense"))
```

```{r isoClassPlot}
col.category <- RColorBrewer::brewer.pal(nlevels(isoClass$structural_category), "Set1")
plot_isoformClass <-
  isoClass %>% filter(count >= 10) %>% ggplot() +
  geom_bar(aes(x = structural_category, fill=structural_category)) +
  theme_bw() +
  # geom_hline(yintercept = 164, linetype="dashed") +
  labs(y = "Number of transcripts", x = "Structural category", fill = "Structural category") +
  theme(text = element_text(size = 20), legend.position = "none",
        axis.text.x = element_text(angle = 30, hjust=1)) +
  scale_fill_manual(values = col.category) +
  scale_y_continuous(labels = scales::comma)
print(plot_isoformClass)

plot_isoformCount <-
  isoClass %>% filter(count >= 10) %>%  dplyr::count(structural_category, wt = count, name = "count") %>% 
  ggplot() +
  geom_bar(aes(x=structural_category, y=count, fill=structural_category), stat = "identity") +
  labs(y = "Read count", x = "Structural category", fill = "Structural category") +
  theme_bw() +
  theme(text = element_text(size = 20), legend.position = "none",
        axis.text.x = element_text(angle = 30, hjust = 1)) +
  scale_fill_manual(values = col.category) +
  scale_y_continuous(labels = scales::unit_format(unit = "M", scale = 1e-6))
print(plot_isoformCount)
```

```{r makeIsoformPlot, echo=FALSE, eval=FALSE}
pdf("plots/isoformClass.pdf", width = 16, height = 6)
plot_grid(plot_isoformClass, plot_isoformCount,
          labels = c("a", "b"), label_size = 20)
dev.off()
```

<!-- ## Short-read junction count -->

<!-- ```{r juncValid} -->
<!-- junc_count <- read.delim("../bambu/sqanti_out/extended_annotations.genes.edited_junctions.txt") -->
<!-- junc_count$structural_category <- isoClass$structural_category[match(junc_count$isoform, isoClass$isoform)] -->
<!-- junc_count -->
<!-- junc_no_dunp <- junc_count %>%  -->
<!--   mutate(junction = paste(chrom, genomic_start_coord, genomic_end_coord, sep = "_")) %>%  -->
<!--   filter(!duplicated(junction)) -->
<!-- ``` -->


## minor intron gene analysis

Check which minor intron genes are not available in our dataset.
```{r checkMiGene}
mi_gene$Gene.Name[!(mi_gene$Gene.Name %in% tmap$ref_gene_id)]
```

Below is the explanation of the *gffcompare* transcript classifications:
![*gffcompare* transcript classifications](gffcompare_codes.png)

```{r qc}
# check how many expressed transcripts were detected
table(rowSums(counts[, c(-1, -2)]) >= 10)
tmap$totalCount <- rowSums(counts[, c(-1, -2)])[match(tmap$qry_id, counts$TXNAME)]
# check the class code of all transcripts
table(tmap$class_code)
# check the class code of expressed transcripts
table(tmap[tmap$totalCount >= 10, "class_code"])
```

We extract transcript class "m" and "n" (intron retentions) from gffcompare output.
```{r extrIsoCls}
tmap.m <- tmap[tmap$class_code == "m", ]
tmap.n <- tmap[tmap$class_code == "n", ]
```

Check how many of "m" and "n" are in the minor intron genes list. We use "ref_gene_id" (annotated gene name) to identify the gene that the isoforms belong to.
```{r ChkIsoCls}
table(tmap.m$ref_gene_id %in% mi_gene$Gene.Name)
table(tmap.n$ref_gene_id %in% mi_gene$Gene.Name)
```

Save the lists of "m" and "n" minor intron isoforms.
```{r SaveIsoCls}
write.csv(tmap.m[tmap.m$ref_gene_id %in% mi_gene$Gene.Name, ],
          "results/miGenesClassMTx.csv")
write.csv(tmap.n[tmap.n$ref_gene_id %in% mi_gene$Gene.Name, ],
          "results/miGenesClassNTx.csv")
```

The transcript ids are in the "qry_id" column of the lists.

[class m](results/miGenesClassMTx.csv)

[class n](results/miGenesClassNTx.csv)

## Differential expression analysis

### Preparing data object

When preparing data objects, we combined counts from quality pass and fail reads for each sample.

```{r DGEList}
x <- DGEList(counts = counts[, c(-1, -2)])
x$counts[, seq(2, 16, 2)] <-x$counts[, seq(1, 15, 2)] + x$counts[, seq(2, 16, 2)] 
x$samples$lib.size[seq(2, 16, 2)] <- x$samples$lib.size[seq(1, 15, 2)] + x$samples$lib.size[seq(2, 16, 2)]
x <- x[, -seq(1, 15, 2)]
x$samples$group <- rep(c("control", "targeted"), c(4, 4))
colnames(x) <- gsub("_pass", "", colnames(x))
x$genes <- tmap[match(rownames(x), tmap$qry_id), 1:6]
x$genes$isMinorIntron <- x$genes$ref_gene_id %in% mi_gene$Gene.Name
```
### Filtering, normalization and MDS plot

It may take a while to load the interactive MDS plot.
```{r preprocess}
filt <- filterByExpr(x)
table(filt)
x <- x[filt, ]
x <- calcNormFactors(x)
x$samples
glimmaMDS(x)
pdf("plots/longDTE/MDS.pdf", height = 5, width = 8)
lcpm <- cpm(x, log=TRUE)
plotMDS(lcpm,
        col = rep(c("#c06636", "#646e3b"), each = 4),
        label = paste(rep(c("NT", "RNPC3"), each = 4), 1:4, sep = "-"),
        main = "ONT long-read MDS plot",
        cex.main = 1.5, cex.lab = 1.5, cex.axis = 1.5, cex = 1.5)
dev.off()
pdf("plots/longDTE/MDS_var.pdf", height = 5, width = 8)
# par(mfrow = c(1, 2))
layout(mat = matrix(1:2, ncol = 2), width = c(1.7, 1))
mds <- plotMDS(lcpm,
        col = rep(c("#c06636", "#646e3b"), each = 4),
        label = paste(rep(c("NT", "RNPC3"), each = 4), 1:4, sep = "-"),
        main = "ONT long-read MDS plot",
        cex.main = 1.3, cex.lab = 1.3, cex.axis = 1.2,
        xlim = c(-2.6, 2.7))
barplot(mds$var.explained[1:7], names.arg = 1:7,
        xlab = "Dimension",
        ylab = "Proportion of variance explained",
        cex.main = 1.3, cex.lab = 1.3, cex.axis = 1.2)
dev.off()
```

On the MDS plot, samples from different groups are separated on dimension 2.

### DE analysis using *edgeR* quasi-likelihood method

```{r DE}
design <- model.matrix(~x$samples$group)
x <- estimateDisp(x, design)
plotBCV(x)
pdf("plots/longDTE/BCV.pdf", height = 5, width = 8)
plotBCV(x,
        cex.main = 1.5, cex.lab = 1.5, cex.axis = 1.5,
        main = "ONT long-read BCV")
dev.off()
# ggplot version of BCV plot
dispersion <- data.frame(
  AveLogCPM = x$AveLogCPM,
  common.dispersion = x$common.dispersion,
  trended.dispersion = x$trended.dispersion,
  tagwise.dispersion = x$tagwise.dispersion
)
dispersion <- dispersion[order(dispersion$AveLogCPM),]
pdf("plots/longDTE/BCV_gg.pdf", height = 5, width = 8)
bcv.plot <- ggplot(dispersion)+
  stat_binhex(aes(x = AveLogCPM, y = sqrt(tagwise.dispersion)), bins = 100) +
  scale_fill_viridis(option = "magma", limits=c(0, 450)) +
  # geom_hline(yintercept = sqrt(x$common.dispersion), aes(colour = "Common")) +
  geom_line(aes(x = AveLogCPM, y = sqrt(common.dispersion), colour = "Common")) +
  geom_line(aes(x = AveLogCPM, y = sqrt(trended.dispersion), colour = "Trend")) +
  scale_colour_manual(name = "BCV",
                      breaks = c("Common", "Trend"),
                      values = c("Common" = "red", "Trend" = "blue")) +
  theme_bw() +
  labs(x = "Average log CPM", y = "Bioligical coefficient of variation",
       fill = "Tagwise BCV\ncount",
       title = "ONT long-read BCV") +
  theme(text = element_text(size = 16))
print(bcv.plot)
dev.off()

qlfit <- glmQLFit(x, design)
plotQLDisp(qlfit)
res <- glmQLFTest(qlfit)
plotMD(res, main = "RNPC3-targeted vs NT")
pdf("plots/longDTE/MD.pdf", height = 5, width = 8)
plotMD(res, 
       cex.main = 1.5, cex.lab = 1.5, cex.axis = 1.5,
       main = "ONT RNPC3-targeted vs NT")
dev.off()
# number of DE transcripts
summary(decideTestsDGE(res))
# top DE transcripts
topTags(res)
tt <- topTags(res, n=Inf)
write.csv(tt, "results/DE_topTags.csv")

# check number of MI transcripts in top n DTEs
top_n_mi <- function(ts){
  res = data.frame(
    n = 1:min(sum(ts$FDR < 0.05), sum(ts$isMinorIntron))
  )
  res$n_mi = sapply(res$n, function(x){
    sum(ts[1:x, "isMinorIntron"])
  }, simplify=TRUE)
  return(res)
}
topN.tt <- top_n_mi(as.data.frame(tt))
pdf("plots/longDTE/topN_MI_DTE.pdf", height = 4)
ggplot(topN.tt, aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DE transcripts", y = "number of minor intron transcripts")
dev.off()
```

```{r eval=FALSE, echo=FALSE}
bcv.plot.short <- readRDS("shortBCV.RDS")
leg <- get_legend(bcv.plot)
pdf("plots/compare/BCV.pdf", height = 5, width = 16)
plot_grid(bcv.plot + 
            theme(legend.position = "none") +
            geom_hline(yintercept = layer_scales(bcv.plot.short)$y$range$range, colour = "grey70"),
          bcv.plot.short + 
            theme(legend.position = "none"),
          leg,
          ncol = 3, rel_widths = c(1, 1, 0.2))
dev.off()
```


[DE results table](results/DE_topTags.csv)

[Number of minor intron genes in top N DE transcripts](plots/longDTE/topN_MI_DTE.pdf)

<!-- It may take a while to load the following interactive mean-difference plot.  -->

<!-- You can click on each dot on the top left plot or each row in the bottom table to check each individual transcript. -->
<!-- You can also search for gene or transcript names in the table. -->
<!-- ```{r glimmaMA} -->
<!-- glimmaMA(res, dge = x, main = "Targeted vs Control") -->
<!-- ``` -->

### Gene set testing

We used the function `fry` from *limma* to test whether specific sets of transcripts are differentially expressed, and barcode plots were made to visualize the results.

The `PValue` column gives the two-sided p-value for testing whether the set is DE as a whole, either up or down. The `PValue.Mixed` column gives a p-value for testing whether genes in the set tend to be DE, without regard to direction. 

#### Minor intron genes from the list provided by Joan.

```{r geneSetMI}
lcpm <- cpm(x, log = TRUE)
fry(lcpm, x$genes$isMinorIntron,design)
barcodeplot(res$table$logFC, index = x$genes$isMinorIntron,
            main = "Minor intron genes, targeted vs control")
pdf("plots/longDTE/barcode_minorIntron.pdf", height = 5, width = 8)
barcodeplot(res$table$logFC, index = x$genes$isMinorIntron,
            main = "ONT minor intron genes, targeted vs control",
            cex.lab = 1.8, cex.axis = 1.5, cex = 1.5)
dev.off()
```
Transcripts from minor intron splicing genes are up regulated.

#### Other lists provided by Joan

Extract table from pdf and add information to`x$genes`
```{r geneLists}
library(tabulizer)
minor_ir <- as.data.frame(extract_tables("../metadata/Short-read splicing analysis.pdf", pages = 1)[[1]])
colnames(minor_ir) <- minor_ir[1 ,]
minor_ir <- minor_ir[-1, ]
colnames(minor_ir)[6:10] <- colnames(minor_ir)[5:9]
colnames(minor_ir)[4] <- "Coord start"
colnames(minor_ir)[5] <- "Coord end"
x$genes$isMinorIR <- x$genes$ref_gene_id %in% minor_ir$`Gene Name`

minor_as <- as.data.frame(extract_tables("../metadata/Short-read splicing analysis.pdf", pages = 2)[[1]])
colnames(minor_as) <- minor_as[1, ]
minor_as <- minor_as[-1, ]
minor_as <- minor_as[, -9]
colnames(minor_as)[5:9] <- colnames(minor_as)[4:8]
colnames(minor_as)[3] <- "Coord start" 
colnames(minor_as)[4] <- "Coord end"
minor_as$gene <- strsplit2(minor_as$`AS event – Gene ID – Gene Name`, "-")[,3]
x$genes$isMinorAS <- x$genes$ref_gene_id %in% minor_as$gene

ir_finder <- as.data.frame(extract_tables("../metadata/Short-read splicing analysis.pdf", pages = 3)[[1]])
colnames(ir_finder) <- ir_finder[1, ]
ir_finder <- ir_finder[-1, ]
x$genes$isIrFinder <- x$genes$ref_gene_id %in% ir_finder$Gene
```

Gene set testing

```{r geneSetLists}
# minor IR
fry(lcpm, x$genes$isMinorIR,design)
barcodeplot(res$table$logFC, index = x$genes$isMinorIR,
            main = "Minor IR top 40 genes, targeted vs control")

# minor AS
fry(lcpm, x$genes$isMinorAS,design)
barcodeplot(res$table$logFC, index = x$genes$isMinorAS,
            main = "Minor AS top 40 gene, targeted vs control")

# IR finder
fry(lcpm, x$genes$isIrFinder,design)
barcodeplot(res$table$logFC, index = x$genes$isIrFinder,
            main = "IR finder unbiased top 40 genes, targeted vs control")
```


#### Transcript classifications

Class "m" transripts
```{r geneSetM}
fry(lcpm, x$genes$class_code=="m", design)
barcodeplot(res$table$logFC, index = x$genes$class_code=="m",
            main = "Transcript class m, targeted vs control")
```


Class "n" transcipts
```{r geneSetN}
fry(lcpm, x$genes$class_code=="n", design)
barcodeplot(res$table$logFC, index = x$genes$class_code=="n",
            main = "Transcript class n, targeted vs control")
```

Class "j" transcripts
```{r geneSetJ}
fry(lcpm, x$genes$class_code=="j", design)
barcodeplot(res$table$logFC, index = x$genes$class_code=="j",
            main = "Transcript class n, targeted vs control")
```

It seems like the expression of transcript classes "m", "n" and "j" are not regulated directionally together.

## Differential transcript usage analysis

### DTU analysis using *limma-diffSplice*

```{r DTU}
v <- voom(x, design, plot = TRUE)
fit <- lmFit(v)
ds <- diffSplice(fit, geneid = x$genes$qry_gene_id, exonid = x$genes$qry_id)
# top DTU genes
topSplice(ds)
# top DTU transcripts
topSplice(ds, test = "t")
ts <- topSplice(ds, test = "simes", number = Inf)
ts.tx <- topSplice(ds, test = "t", number = Inf)
# number of DTU genes
table(ts$FDR < 0.05)
# number of DTU transcripts
table(ts.tx$FDR < 0.05)

write.csv(ts, file = "results/DTU_gene_topSplice.csv")
write.csv(ts.tx, file = "results/DTU_transcript_topSplice.csv")

# check number of top DTU MI genes
topN.ts <- top_n_mi(ts)
pdf("plots/topN_MI_DTU.pdf", height = 5, width = 8)
ggplot(topN.ts, aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DTU genes", y = "number of minor intron genes")
dev.off()

topN.ts.tx <- top_n_mi(ts.tx)
pdf("plots/topN_MI_DTU_tx.pdf", height = 5, width = 8)
ggplot(topN.ts.tx, aes(x = n, y = n_mi)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  labs(x = "top n DTU transcripts", y = "number of minor intron transcripts")
dev.off()
```

[DTU gene-level results table](results/DTU_gene_topSplice.csv)

[DTU transcript-level results table](results/DTU_transcript_topSplice.csv)

[Number of minor intron genes in top N DTU genes](plots/topN_MI_DTU.pdf)

[Number of minor intron transcripts in top N DTU transcripts](plots/topN_MI_DTU_tx.pdf)

### Gene set testing based on transcript-level DTU results

<!-- In the output, significant "Down" and "Up" p-value means the DE results are negatively or positively correlated with the expression of the set. Significant "UpOrDown" p-value means up or down regulation, but the direction is not specified. "Mixed" is used to test the correlation without regard for direction. -->

#### Minor intron genes

```{r geneSetDTUMI}
barcodeplot(ds$t[, 2], index = ds$genes$isMinorIntron,
            main = "Minor intron genes, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$isMinorIntron, ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$isMinorIntron, ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$isMinorIntron, ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```

#### Other lists provided by Joan

```{r geneSetDTUminorIR}
barcodeplot(ds$t[, 2], index = ds$genes$isMinorIR,
            main = "Minor IR top 40 genes",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$isMinorIR, ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$isMinorIR, ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$isMinorIR, ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```

```{r geneSetDTUminorAS}
barcodeplot(ds$t[, 2], index = ds$genes$isMinorAS,
            main = "Minor AS top 40 genes",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$isMinorAS, ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$isMinorAS, ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$isMinorAS, ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```

```{r geneSetDTUirFinder}
barcodeplot(ds$t[, 2], index = ds$genes$isIrFinder,
            main = "IR finder top 40 genes",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$isIrFinder, ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$isIrFinder, ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$isIrFinder, ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```

#### Transcript classifications

Class "m" transripts
```{r geneSetDTUM}
barcodeplot(ds$t[, 2], index = ds$genes$class_code=="m",
            main = "Transcript class m, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$class_code=="m", ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$class_code=="m", ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$class_code=="m", ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```


Class "n" transcipts
```{r geneSetDTUN}
barcodeplot(ds$t[, 2], index = ds$genes$class_code=="n",
            main = "Transcript class n, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$class_code=="n", ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$class_code=="n", ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$class_code=="n", ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```

Class "j" transcripts
```{r geneSetDTUJ}
barcodeplot(ds$t[, 2], index = ds$genes$class_code=="j",
            main = "Transcript class j, targeted vs control",
            xlab = "t-statistics")
test.mixed <- fisher.test(table(ts.tx$class_code=="j", ts.tx$FDR < 0.05))
test.mixed$p.value
test.up <- fisher.test(table(ts.tx$class_code=="j", ts.tx$FDR < 0.05 & ts.tx$logFC > 0))
test.up$p.value
test.down <- fisher.test(table(ts.tx$class_code=="j", ts.tx$FDR < 0.05 & ts.tx$logFC < 0))
test.down$p.value
```


## Visualization of transcripts coverage

Due to high computation load, visualization was done in a separate script. notice that genes without known strand information are not plotted.

```{r saveGeneName}
# save DTE gene names
tt <- as.data.frame(tt)
saveRDS(unique(tt$qry_gene_id[tt$FDR < 0.05 & tt$isMinorIntron]), "rds/DTEgenesMI.RDS")
saveRDS(unique(tt$qry_gene_id[tt$FDR < 0.05 & !tt$isMinorIntron]), "rds/DTEgenesNotMI.RDS")
saveRDS(unique(ts$qry_gene_id[ts$FDR < 0.05 & ts$isMinorIntron]), "rds/DTUgenesMI.RDS")
saveRDS(unique(ts$qry_gene_id[ts$FDR < 0.05 & !ts$isMinorIntron]), "rds/DTUgenesNotMI.RDS")
```

```{r txViz, eval=FALSE}
source("plot_cov.R")
```

[DTE minor intron genes](plots/cov_DTEgenesMI2.pdf)

[DTE not minor intron genes](plots/cov_DTEgenesNotMI2.pdf)

[DTU minor intron genes](plots/cov_DTUgenesMI2.pdf)

[DTU not minor intron genes](plots/cov_DTUgenesNotM2I.pdf)

[Minor intron genes that are not DTE or DTU but containing class m, n or j transcripts](plots/cov_notSigMI_mnj.pdf)


## To do list

1. Quality control plots for this data set



3. GO/KEGG analysis (the `goana` and `kegga` functions are only available for gene level)

## Session information
```{r sessionInfo}
sessionInfo()
```
